python/paddle/fluid/imperative/checkpoint.py

```diff
@@ -0,0 +1,138 @@
+            x_data = x_data.reshape((-1, num_steps, 1))
+            y_data = y_data.reshape((-1, 1))
+            init_hidden_data = np.zeros(
+                (num_layers, batch_size, hidden_size), dtype='float32')
+            init_cell_data = np.zeros(
+                (num_layers, batch_size, hidden_size), dtype='float32')
+            x = to_variable(x_data)
+            y = to_variable(y_data)
+            init_hidden = to_variable(init_hidden_data)
+            init_cell = to_variable(init_cell_data)
+            dy_loss, last_hidden, last_cell = ptb_model(x, y, init_hidden,
+                                                        init_cell)
+            param_path = "./my_paddle_model"
+            fluid.imperative.save_persistables(dirname=param_path,
+                                       layer=ptb_model)
+    """
+    # Todo: PyLayer
+    if isinstance(layer, Layer):
+        layers_parameter_dict = {}
+        for param in layer.parameters():
+            layers_parameter_dict[param.name] = param._numpy()
+        _save_var_to_file(layers_parameter_dict, dirname, filename)
+
+
+def load_persistables(dirname, filename=None):
+    """
+    This function trys to load persistable variables from the folder
+    `dirname` or the file `filename`.
+
+    Use the `dirname` to specify the folder where persistable variables were
+    saved. If variables were saved in separate files, set `filename` None;
+    if all variables were saved in a single file, use `filename` to specify
+    the file name.
+
+    Args:
+        dirname(str): The directory path.
+        filename(str|None): The file which saved all variables, this file path should be end with '.npz'. If variables were
+                            saved in differnet files, set it to None.
+                            Default: None
+
+    Returns:
+        dict: The parameter-dict resumed from file
+
+    Examples:
+        .. code-block:: python
+
+            param_path = "./my_paddle_model"
+            param_dict = fluid.imperative.load_persistables(param_path)
+            param_1 = param_dict['PtbModel_0.w_1']
+
+            or:
+            param_path = "./my_paddle_model"
+            filename = "model.npz"
+            param_dict = fluid.imperative.load_persistables(param_path, filename=filename)
+            param_1 = param_dict['PtbModel_0.w_1']
+
+        """
+    parameter_dict = {}
+    if filename is not None:
+        # todo lujun: if remove np.save/load, fix this
+        parameter_dict = numpy.load(os.path.join(dirname, filename))
+    else:
+        for pwd, dirs, files in os.walk(dirname):
+            for parameter_file in files:
+                # todo lujun: if remove np.save/load, fix this
+                if os.path.splitext(parameter_file)[-1] == '.npy':
+                    parameter_dict[parameter_file[:-4]] = numpy.load(
+                        os.path.join(pwd, parameter_file))
+    return parameter_dict
```

Comment:
how to use this dict?

Reply:
it's a Variable dict. like : `param_dict['PtbModel_0.w_1']`. The key of the dictionary is the parameter name when it is saved( no full_name). as the file name :
`PtbModel_0.w_0
PtbModel_0.w_1
embedding_para`